web/src/people.ts

```diff
@@ -1194,9 +1195,12 @@ export function build_termlet_matcher(termlet: string): (user: User) => boolean
 
     return function (user: User): boolean {
         let full_name = user.full_name;
+        // Only ignore diacritics if the query is plain ascii
         if (is_ascii) {
-            // Only ignore diacritics if the query is plain ascii
-            full_name = typeahead.remove_diacritics(full_name);
+            if (user.stripped_full_name === undefined) {
+                user.stripped_full_name = typeahead.remove_diacritics(full_name);
+            }
+            full_name = user.stripped_full_name;
         }
         const names = full_name.toLowerCase().split(" ");
```

Comment:
I think `stripped_full_name` is also a confusing variable name; `name_with_diacritics_removed` would be reasonable. 

And as Anders requested, this field has to be set back to `undefined` in the live-update code in the `user_events` module, or the matching will be wrong after a user's name changes.

Reply:
Ah right sorry for forgetting that. Making those updates now!